src/hotspot/share/runtime/osThread.cpp

```diff
@@ -38,7 +38,7 @@ OSThread::~OSThread() {
 
 // Printing
 void OSThread::print_on(outputStream *st) const {
-  st->print("nid=0x%x ", thread_id());
+  st->print("nid=" UINT64_FORMAT " ", (uint64_t)thread_id());
```

Comment:
Why are you forcing this to be a 64-bit type?

Reply:
Thomas suggested using UINT64_FORMAT rather than `%d`:
> You'd do:

> print("nid: " UINT64_FORMAT, (uint64_t) id):;

> thread_t is, among other things, pthread_t, which is opaque. Any current code treating that as signed int is incorrect too.

There is no uniform format for the formatted output of thread_id in hotspot. As far as I can see, `%ld` `%d` and `UINTX_FORMAT` are used, so I want to left the decision to reviewers. But IMHO, I prefer using `%d` since a large portion of existing code using `%d`.